src/ConnectedMode/BranchMatcher.cs

```diff
@@ -113,19 +113,19 @@ private async Task<string> DoGetMatchingBranch(string projectKey, IRepository gi
             {
                 var localBranch = gitRepo.Branches.FirstOrDefault(r => string.Equals(r.FriendlyName, remoteBranch.Name, StringComparison.InvariantCultureIgnoreCase));
 
-                if (localBranch == null) { continue; }
+                if (localBranch is null) { continue; }
 
                 var distance = GetDistance(headCommits.Value, localBranch, closestDistance);
 
-                if (distance < closestDistance)
+                if (distance < closestDistance || (distance < int.MaxValue && distance == closestDistance && remoteBranch.IsMain))
```

Comment:
added "distance < int.MaxValue" so it wouldn't automatically match the main branch in this step. We eventually match it but the log message would be wrong.